### PR TITLE
Add docs about integration operator metrics and Prometheus

### PR DESCRIPTION
**Pull request by nikita-vanyasin**
- standard Prometheus installation (no-TLS)
- standard Prometheus installation (TLS)
- Prometheus Operator installation

### CODE REVIEW

File: docs/design/metrics.md [hunk 1]
```diff
@@ -1,12 +1,59 @@
...
+After creating operator deployment, you must configure Prometheus using a configuration file that instructs it
+about which targets to scrape.
+To do so, add a new scrape job to your prometheus.yaml config:
+```yaml
+scrape_configs:
+  - job_name:       'arangodb-operator'
+
+    scrape_interval: 10s # scrape every 10 seconds.
+
+    scheme: 'https'
```
Comment (ajanikow) [inline docs/design/metrics.md:25]: When no TLS then why the scheme is defined as HTTPS?

Default Prometheus installation does not support custom CAs, please check `spec.metrics.tls: false` for this.

Also, there are missing annotations used by prometheus.

File: docs/design/metrics.md [hunk 2]
```diff
@@ -1,12 +1,59 @@
...
+    tls_config:
+      insecure_skip_verify: true
+
+    static_configs:
+      - targets:
+          - "<operator-endpoint-ip>:8528"
+```
+
+
+## ArangoDB metrics
```
Comment (ajanikow) [inline docs/design/metrics.md:35]: There is a collision with PR https://github.com/arangodb/kube-arangodb/pull/1041/files - please, merge it in one if you modify same file

**Review by ajanikow — CHANGES_REQUESTED** [review]

File: docs/design/metrics.md (hunk 2 above)
Comment (nikita-vanyasin) [reply docs/design/metrics.md:67]: Merged into one PR (this)

File: docs/design/metrics.md (hunk 1 above)
Comment (nikita-vanyasin) [reply docs/design/metrics.md:26]: @ajanikow 
Current operator server runs only on HTTPS protocol.
If you request with HTTP proto you will get message `Client sent an HTTP request to an HTTPS server.`

`insecure_skip_verify: true` in Prometheus config allows to fetch data with no problems. Tested with latest stable `prom/prometheus` image

Comment (nikita-vanyasin) [reply docs/design/metrics.md:26]: `spec.metrics.tls` applies to ArangoDB sidecard-exporters, which can use HTTP

Comment (nikita-vanyasin) [reply docs/design/metrics.md:27]: Added info on TLS-enabled connection

**Review by ajanikow — APPROVED** [review]

**Pull request merged**